build.py

```diff
@@ -275,27 +285,36 @@ def clone_or_update_repo(name, branch, dir_name, owner='openwisp', dest=None):
 
     if os.path.exists(clone_path):
         print(f"Repository '{name}' already exists. Updating...")
-        subprocess.run(['git', 'checkout', branch], cwd=clone_path, check=True)
+        # "-c advice.detachedHead=false" is used to suppress the warning
+        # about being in a detached HEAD state when checking out tags.
+        subprocess.run(
+            ['git', '-c', 'advice.detachedHead=false', 'checkout', branch],
+            cwd=clone_path,
+            check=True,
+        )
+        # In CI/production environments, the repository is always freshly
+        # cloned, so pulling is unnecessary and would only result in redundant
+        # network calls.
+        # During local development, we attempt to pull updates, but only if the
+        # current HEAD is on a branch (i.e., not detached, such as when on a tag).
+        if not os.environ.get('PRODUCTION', False) and git_is_on_branch(clone_path):
```

Comment:
`not os.environ.get('PRODUCTION', False)` with this condition, we saved close to 2 minutes on the CI job. 

[CI run without the condition](https://github.com/openwisp/openwisp-docs/actions/runs/14570791947) - 15m 37s
[CI run with the condition](https://github.com/openwisp/openwisp-docs/actions/runs/14571137445) - 13m 47s